keanu-project/src/main/java/io/improbable/keanu/util/dot/WriteDot.java

```diff
@@ -0,0 +1,279 @@
+        }
+    }
+
+
+    // Utility function for forming a DOT string to represent an edge.
+    private static String getEdgeString(VertexId v1, VertexId v2) {
+        if (v1.compareTo(v2) < 0) {
+            return  "<" + v1.hashCode() + "> -> <" + v2.hashCode() + ">";
+        }
+        else {
+            return  "<" + v2.hashCode() + "> -> <" + v1.hashCode() + ">";
+        }
+    }
+
+
+    /**
+     * Creates and stores a label for the given vertex.
+     * Label contains the information that will be displayed for this vertex, such as vertex class or display name and value.
+     *
+     * @param v a vertex
+     */
+    private static void setLabel(Vertex v){
+
+        String vertexLabel;
+
+        // If vertex is a constant vertex, display only it's value.
+        if ((vertexLabel = getValueAsString(v)).equals("")) {
+            // Else use vertex label as DOT label for it.
+            DotAnnotation vertexAnnotation = v.getClass().getAnnotation(DotAnnotation.class);
+            if (v.getLabel() != null) {
+                vertexLabel = v.getLabel().getUnqualifiedName();
+            }
+            // If label isn't set, and if DOT annotation is added to this class, use display name for label.
+            else if (vertexAnnotation != null && !vertexAnnotation.displayName().equals("")) {
+                vertexLabel = vertexAnnotation.displayName();
+            }
+            // Else use class name.
+            else {
+                vertexLabel = v.getClass().getSimpleName();
+            }
+        }
+
+        int uniqueHiddenID = v.getId().hashCode();
+        String fullName = uniqueHiddenID + "[label=\"" + vertexLabel + "\"]";
+        idsToLabelStrings.put(uniqueHiddenID, fullName);
+    }
+
+
+    // Get value of constant vertices that always have their value set.
+    private static String getValueAsString(Vertex v) {
+        if (v.getClass().equals(ConstantDoubleVertex.class)) {
+            return "" + ((ConstantDoubleVertex) v).getValue().getValue(0);
+        }
+        if (v.getClass().equals(ConstantIntegerVertex.class)) {
+            return "" + ((ConstantIntegerVertex) v).getValue().getValue(0);
+        }
+        if (v.getClass().equals(ConstantBoolVertex.class)) {
+            return "" + ((ConstantBoolVertex) v).getValue().getValue(0);
+        }
+        return "";
+    }
+
+
+    // Apply hyperparameters as a label to connection edges.
+    // Note - this is currently only done for Gaussian vertices. Can be extended for any others.
+    private static void applyHyperparameterInfo(Vertex v) {
+        VertexId vId = v.getId();
+
+        if (v.getClass().getSimpleName().equals(GaussianVertex.class.getSimpleName())) {
+            VertexId meanVertexId = ((GaussianVertex) v).getMu().getId();
+            verticeIdsToConnectionInfo.put(concatenateVertexIds(vId, meanVertexId), verticeIdsToConnectionInfo.get(concatenateVertexIds(vId, meanVertexId)) + " [label=mu]");
```

Comment:
`verticeIdsToConnectionInfo.computeIfPresent(concatenateVertexIds(vId, meanVertexId), (ids, previousValue) -> previousValue + " [label=mu]");`

Reply:
Oh, that's cool!